test/unit/isolated/config.test.ts

```diff
@@ -28,6 +30,26 @@ describe('config', () => {
       const config = await loadConfig(PHASE_DEVELOPMENT_SERVER, pathToConfigFn)
       expect(config.distDir).toEqual('.next')
       expect(config.onDemandEntries.maxInactiveAge).toBeDefined()
+      expect(config.images).toEqual(imageConfigDefault)
+    })
+
+    it('Should prefix base path to images path', async () => {
+      const config = await loadConfig(
+        PHASE_DEVELOPMENT_SERVER,
+        pathToBasePathConfig
+      )
+      expect(config.images.path).toBe(`/docs${imageConfigDefault.path}`)
+    })
+
+    it('Should be an idempotent function', async () => {
```

Comment:
Maybe the entire loadConfig() function should be memoized instead of cloning

Reply:
I typically use factory functions to avoid these issues.

```ts
const defaultConfig = () => {
    return {
        foo: "bar"
    }
}
```

Now I get a new defaultConfig every time!

Memoizing comes with complexity of determining cache keys, and TTL. Feels a bit overkill over just brining in a deep clone or moving to factory functions.

Reply:
What about only calling loadConfig() once instead of multiple times?

Reply:
The Nx team is using it to load the config ahead of constructing a new `NextServer`. I've shared that in the [issue](https://github.com/vercel/next.js/issues/32430#issuecomment-992547625).

Not calling it twice doesn't change the fact that `images` and `imageConfigDefault ` are the same object memory reference on this [line](https://github.com/vercel/next.js/blob/v12.0.7/packages/next/server/config.ts#L313). That's certainly not the intended check because that will always be true. They are the same object.

Reply:
I think the factory functions is the best solution here because it avoids cloning. Something like `getImageConfigDefault()` 👍

Reply:
I dig it! Would you prefer an update to this PR with that approach, or see it in a new PR to compare?

Reply:
Either way is fine

Reply:
https://github.com/vercel/next.js/pull/32474